ENIGMAsystem/SHELL/Platforms/Cocoa/CocoaMain.cpp

```diff
@@ -32,33 +32,102 @@
 #include "../General/PFfilemanip.h"
 
 #include "Platforms/General/PFmain.h"
+#include "Platforms/General/PFfilemanip.h"
 #include "Universal_System/roomsystem.h"
 #include "Universal_System/estring.h"
 
+/* Uncomment the 2 lines below for testing generate_working_directory() success/failure */
+// #include "Widget_Systems/widgets_mandatory.h"
+// #include "Widget_Systems/Cocoa/dialogs.h"
+
 using std::string;
 
+using enigma_user::set_working_directory;
+using enigma_user::directory_exists;
+using enigma_user::filename_name;
+using enigma_user::filename_dir;
+using enigma_user::filename_path;
+using enigma_user::filename_ext;
+
 namespace enigma {
 
 void SetResizeFptr();
 
+static inline void generate_working_directory() {
```

Comment:
dafux are you "generating" ?

Reply:
MacOS apps are directories that store your executable, the app icon, and various dependencies. Read here for more info: https://www.maketecheasier.com/anatomy-macos-app-bundles/ this code replaces the original Objective-C solution to set the working directory correctly to the app bundle's resources folder if a correctly structured app bundle in fact exists, otherwise use the default POSIX cwd.

Reply:
right. youre setting some stuff but afaik youre "generating" nothing, This feels incorrectly named